src/hardware/uni-t-ut8802e/protocol.c

```diff
@@ -0,0 +1,430 @@
+/*
+ * This file is part of the libsigrok project.
+ *
+ * Copyright (C) 2023 Konrad Tagowski <konrad.tagowski@grinn-global.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <config.h>
+#include "math.h"
+
+#include "protocol.h"
+
+/*
+ * Developer notes on the UT8802E protocol:
+ * - Serial communication over HID CP2110 converter USB-UART.
+ * - UART frame format 8n1 at 9600 bps.
+ * - DMM packet starts with a magic marker, functionality, value, etc.
+ * - UT8802E only sends a packet with measurement. Example of frame:
+ *   Frame | func | value    | comma | settings | checksum |
+ *   ac    | 1b   | 45 01 00 | 33	 | 04  	    | 44       |
+ * - The frame to send a command to the multimeter is unknown.
+*/
```

Comment:
This really belongs in the wiki, not in the code. Can you please create a page for this hardware? @abraxa will hook you up with an account. Document as much detail of the protocol as you can.